From 8fdcfc9b4b22778d3e24c8f6cab81dbce2d4c3e3 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@apache.org>
Date: Wed, 20 Apr 2016 23:40:46 -0700
Subject: [PATCH 1437/2848] YARN-4935. TestYarnClient#testSubmitIncorrectQueue
 fails when FairScheduler is the default. (Yufei
 Gu via kasha)

(cherry picked from commit 6e297836d64498d90d0cd343789db2f2d490378a)
(cherry picked from commit cd74cf9525ebba263b26593b2dd2bf6145ed4541)

Change-Id: I7cf2ef9f7eb61e554618c9a756eb920128729644
---
 .../yarn/client/api/impl/TestYarnClient.java       |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestYarnClient.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestYarnClient.java
index 82cb9fc..f25dce2 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestYarnClient.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/api/impl/TestYarnClient.java
@@ -193,12 +193,15 @@ public void testSubmitApplication() {
   }
 
   @Test (timeout = 30000)
-  public void testSubmitIncorrectQueue() throws IOException {
+  public void testSubmitIncorrectQueueToCapacityScheduler() throws IOException {
     MiniYARNCluster cluster = new MiniYARNCluster("testMRAMTokens", 1, 1, 1);
     YarnClient rmClient = null;
     try {
-      cluster.init(new YarnConfiguration());
-	     cluster.start();
+      YarnConfiguration conf = new YarnConfiguration();
+      conf.set(YarnConfiguration.RM_SCHEDULER,
+        CapacityScheduler.class.getName());
+      cluster.init(conf);
+      cluster.start();
       final Configuration yarnConf = cluster.getConfig();
       rmClient = YarnClient.createYarnClient();
       rmClient.init(yarnConf);
-- 
1.7.9.5

